### PR TITLE
feat: Added forecasting snippets and fixed bugs with existing snippets

### PR DESCRIPTION
**Pull request by ivanmkc**
# Issues with existing snippets fixed
- [x] dataset_id should be a `str` and not an `int`
- [x] create_training_pipeline_tabular_classification_sample and create_training_pipeline_tabular_regression_sample are missing parameters
- [x] Renamed samples/model-builder/explain_tabular_sample.py to samples/model-builder/explain_sample.py since it is universally applicable.

# Added snippets and accompanying tests
- [x] samples/model-builder/create_and_import_dataset_time_series_bigquery_sample.py
- [x] samples/model-builder/create_and_import_dataset_time_series_gcs_sample.py
- [x] samples/model-builder/create_training_pipeline_forecasting_sample.py
- [x] samples/model-builder/create_batch_prediction_job_dedicated_resources_bigquery_sample.py
- [x] samples/model-builder/create_batch_prediction_job_sample_bigquery.py

### CODE REVIEW

File: samples/model-builder/create_training_pipeline_forecasting_sample.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        data_granularity_count=data_granularity_count,
+        training_fraction_split=training_fraction_split,
+        validation_fraction_split=validation_fraction_split,
+        test_fraction_split=test_fraction_split,
+        budget_milli_node_hours=budget_milli_node_hours,
+        model_display_name=model_display_name,
+        sync=sync,
+    )
+
+    model.wait()
```
Comment (TheMichaelHu) [inline samples/model-builder/create_training_pipeline_forecasting_sample.py:70]: If we're always going to call `wait()` right after, why not just always have `sync=True` and remove sync as an arg?

Comment (ivanmkc) [reply samples/model-builder/create_training_pipeline_forecasting_sample.py:70]: Thanks Michael, I will add the other args that are important. 

Let me double-check about sync, as I'm following the common convention for other samples.

File: samples/model-builder/create_training_pipeline_forecasting_sample_test.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    mock_get_time_series_dataset.assert_called_once_with(constants.RESOURCE_ID)
+
+    mock_sdk_init.assert_called_once_with(
+        project=constants.PROJECT, location=constants.LOCATION
+    )
+    mock_get_automl_forecasting_training_job.assert_called_once_with(
+        display_name=constants.DISPLAY_NAME,
+        optimization_objective="minimize-rmse",
+    )
+    mock_run_automl_forecasting_training_job.assert_called_once_with(
```
[the remark this answers is not in the document]
Comment (ivanmkc) [reply samples/model-builder/create_training_pipeline_forecasting_sample_test.py:49]: I don't think so. That is a mock with no run method defined.

I'm just following the convention with the other test by the way.

File: samples/model-builder/conftest.py [hunk 3]
```diff
@@ -136,6 +158,12 @@ def mock_import_tabular_dataset(mock_tabular_dataset):
         yield mock
 
 
+@pytest.fixture
+def mock_import_time_series_dataset(mock_time_series_dataset):
+    with patch.object(mock_time_series_dataset, "import_data") as mock:
+        yield mock
```
[the remark this answers is not in the document]
Comment (ivanmkc) [reply samples/model-builder/conftest.py]: Good catch, I think I can remove this.

File: samples/model-builder/create_and_import_dataset_time_series_bigquery_sample.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+):
+
+    aiplatform.init(project=project, location=location)
+
+    dataset = aiplatform.TimeSeriesDataset.create(
+        display_name=display_name,
+        bq_source=bq_source,
+    )
+
+    dataset.wait()
```
[the remark this answers is not in the document]
Comment (TheMichaelHu) [reply samples/model-builder/create_and_import_dataset_time_series_bigquery_sample.py:34]: Either this or I think you would want to add `sync=sync` like in the samples for training and prediction.

**Review by reznitskii — APPROVED** [review]

**Pull request merged**